### ISSUE TITLE
Move config folder to $XDG_CONFIG_HOME

### ISSUE THREAD

**Issue by ZJvandeWeg** [sup-heliotrope/sup#398]
To unclutter everyone's $HOME dir, I would propose usage of the system var XDG_CONFIG_HOME (~/.config usually)
Or to be able to start sup with;
`sup --config ~/.config/sup` so `alias sup=sup --config ~/.config/sup` would be available.

Would this break sup for any non-unix users?

**Comment by rakoo** [comment]
:+1: on the idea. The hardest part will be managing transition:

* if `~/.config/sup` exists
  * If `~/.sup` doesn't exist, use `~/.config/sup`, return
  * if `~/.sup` exists, put some warning saying that it will be discarded and any setting of interest should be moved to `~/.config/sup`, return
* Else if `~/.config/sup` doesn't exist, move `~/.sup` to `~/.config/sup`, remove `~/.sup` and load from new dir

Instead of using `--config`, you can use the `SUP_BASE` environment variable today already; I don't think it's worth it to add yet another parameter just for that.

I have no idea how non-unix works at this level, unfortunately.

**Comment by ZJvandeWeg** [comment]
Well to be honest, I've used dozens of applications which have multiple possible locations for their config files and such. I suggest the same for sup. First checking SUP_BASE, then XDG_CONFIG_HOME/sup and last ~/.sup.

Implementing this would be just two times this;
```ruby
# Written without editor, might contain syntax errors etc
BASE_DIR = if ENV["SUP_BASE"] then ENV["SUP_BASE"]
                    elsif ENV["XDG_CONFIG_HOME"] then File.join(ENV['XDG_CONFIG_HOME', 'sup')
                    else File.join(ENV['HOME']], '.sup')
                    end
```

**Comment by rakoo** [comment]
Yes sure, but what we want is also people migrating to the new `$XDG_CONFIG_HOME/.sup` folder, and I like that my applications "automagically" migrate their own conf to their new format/folder without me having to read the latest CHANGELOG.

Moreover, keeping a simple `BASE_DIR` would be attractive:

```ruby
BASE_DIR = ENV["SUP_BASE"] || ENV["XDG_CONFIG_HOME"] and File.join(ENV["XDG_CONFIG_HOME"], "sup")
```

and then, just before loading the conf, checking if the user still has the old `~/.sup` and take appropriate measures, like moving the folder or even crashing with an informative message: "Sup's config has been moved to `$XDG_CONFIG_HOME/.sup`, please move your settings over there".

In short, I want to handle a single official dir, not 2 that may be living until who knows when.

**Comment by ZJvandeWeg** [comment]
This implies that all systems have the var set? Not sure if that's the case, from the top of my mind it isn't for ArchLinux. 
But for support reasons and the xapian dir I see your point.

Matthieu Rakotojaona <notifications@github.com> schreef op 20 februari 2015 21:30:14 CET:

**Comment by gauteh** [comment]
Following this scheme, the xapaian index would have to go in
XDG_DATA_HOME.

**Comment by ZJvandeWeg** [comment]
You are right. :+1:

**Comment by gauteh** [comment]
The vars have defaults, in case the envs are not set: http://standards.freedesktop.org/basedir-spec/basedir-spec-latest.html

**Comment by maxmeyer** [comment]
BTW:
```
$XDG_CONFIG_HOME/.sup -> $XDG_CONFIG_HOME/sup
```

**Comment by ZJvandeWeg** [comment]
Im aware, still thanks though. Will create a PR this week.

**Comment by gauteh** [comment]
Closed in #401.
